dbt/adapters/athena/impl.py

```diff
@@ -216,33 +218,36 @@ def apply_lf_grants(self, relation: AthenaRelation, lf_grants_config: Dict[str,
 
     @available
     def is_work_group_output_location_enforced(self) -> bool:
-        conn = self.connections.get_thread_connection()
-        creds = conn.credentials
-        client = conn.handle
+        if self.work_group_output_location_enforced is None:
```

Comment:
let's leave a comment why we're doing a cache here: to avoid rate limiting in case of large dbt projects